File: harness/lua/tests/test_native_build.lua
local t = require("support.assertions")
local c_api = require("support.c_api")
local native = require("polyglot_native")

t.equal(native.release, "Lua 5.5.0")
t.equal(native.add(20, 22), 42)
t.matches(c_api.run("state"), "C API case passed: state")
t.matches(c_api.run("version-gc"), "C API case passed: version%-gc")

-- 该测试证明锁定头文件、liblua、宿主和动态模块能够一起严格构建并执行。
t.done()

File: harness/ruby/tests/test_native_build.rb
# frozen_string_literal: true

require "assertions"
require "polyglot_native"
require "rbconfig"

A = PolyglotAssertions

A.case("locked CRuby headers and shared-library suffix are available to extconf") do
  header_directory = RbConfig::CONFIG.fetch("rubyhdrdir")
  architecture_header = RbConfig::CONFIG.fetch("rubyarchhdrdir")

  A.equal("/opt/polyglot/ruby-4.0.6/include/ruby-4.0.0", header_directory)
  A.truth(File.file?(File.join(header_directory, "ruby.h")))
  A.truth(File.file?(File.join(header_directory, "ruby", "thread.h")))
  A.truth(File.file?(File.join(architecture_header, "ruby", "config.h")))
  A.equal("so", RbConfig::CONFIG.fetch("DLEXT"))
end

A.case("the repository extension builds and exposes its public Ruby API") do
  A.equal("CRuby 4.0.6", PolyglotNative::RELEASE)
  A.equal(42, PolyglotNative.add(20, 22))
end

A.done
